tests/test_tools.py

```diff
@@ -946,24 +946,27 @@ def test_env_path_add_replace_no_dupes_front_replace_existing():
     # Test replaces without dupes when added to front when adding existing entry
     path = EnvPath(["/home/wakka", "/home/wakka/bin"])
     path.add("/home/wakka/bin", front=True, replace=True)
-    exp = ["/home/wakka/bin", "/home/wakka"]
-    assert exp == path
+    assert path == ["/home/wakka/bin", "/home/wakka"]
 
 
 def test_env_path_add_replace_no_dupes_front_replace_multiple():
     # Test replaces without dupes when added to front when multiple existing occurrences
     path = EnvPath(["/home/wakka", "/home/wakka/bin", "/home/wakka/bin"])
     path.add("/home/wakka/bin", front=True, replace=True)
-    exp = ["/home/wakka/bin", "/home/wakka"]
-    assert exp == path
+    assert path == ["/home/wakka/bin", "/home/wakka"]
 
 
 def test_env_path_add_replace_no_dupes_back_replace_multiple():
     # Test replaces without dupes when not added to front
     path = EnvPath(["/home/wakka", "/home/wakka/bin", "/home/wakka/bin"])
     path.add("/home/wakka/bin", front=False, replace=True)
-    exp = ["/home/wakka", "/home/wakka/bin"]
-    assert exp == path
+    assert path == ["/home/wakka", "/home/wakka/bin"]
+
+
+def test_env_path_add_pathlib():
+    path = EnvPath(["/home/wakka", "/home/wakka/bin", "/home/wakka/bin"])
+    path.add(pathlib.Path("/home/wakka/bin"), front=False, replace=True)
+    assert path == ["/home/wakka", "/home/wakka/bin"]
```

Comment:
"Add a test with pathlib.Path("/home/wakka/bin") to make sure that works correctly (which is the original issue)"
Added test here as requested.